templates/vue/src/store/getters.js

```diff
@@ -93,27 +93,37 @@ export function isVisible(state, { getNode, hasMultiContentAncestor }) {
   }
 }
 
-export function getActivities(state) {
-  return (options = {}) => {
-    const { exclude = [] } = options
-    return Object.values(state.nodes)
-      .filter(node => !exclude.includes(node.id) && Boolean(node.quiz))
-      .flatMap(node => node.quiz)
-  }
-}
-
 export function getQuestion(state) {
   return id => {
     const node = Object.values(state.nodes)
-      .filter(node => node.quiz)
-      .find(node => node.quiz.find(q => q.id == id))
+      .filter(node => node.typeData.activity)
+      .find(node => node.typeData.activity.questions.find(q => q.id == id))
     if (node) {
-      return node.quiz.find(q => q.id == id)
+      return node.typeData.activity.questions.find(q => q.id == id)
     }
     return null
   }
 }
 
+export function getAnswers(state) {
+  return (nodeId, questionId) => {
+    if (state.userAnswers.hasOwnProperty(nodeId)) {
+      if (state.userAnswers[nodeId].hasOwnProperty("activity")) {
+        if (state.userAnswers[nodeId].activity.hasOwnProperty(questionId)) {
+          if (
+            typeof state.userAnswers[nodeId].activity[questionId].answers !==
+            "string"
+          )
+            return state.userAnswers[nodeId].activity[questionId].answers
+        }
+      }
+    }
+    return {
+      text: "",
+    }
```

Comment:
How come this includes text entry but not audio? Do we even need the text entry?

Reply:
Hi @wynnset, the audio is retrieving the audio answer from the backend in a different way and does not use this default value. The TextQuestion component accepts an object with a text property as a prop and this object serves as a default value when there is no answer in the backend for this activity yet.

Reply:
So if there is no answer in the backend, we are assuming that the user has an empty string as answer to the the text answer type? Why not just remove it? That's confusing.